### ISSUE TITLE
Warning about TypeVar when bound byProtocol

### ISSUE THREAD

**Issue by imLew** [microsoft/pyright#5595]
**Describe the bug**
When a `TypeVar` is bound by a `Protocol` class is used as a type in a function signature pyright issues a warning.
I don't think this should happen as there is no equivalent to  a `TypeVar` with a bound.

**Code or Screenshots**
Code:
```python
from typing import Protocol, TypeVar

class SupportsSomeMethod(Protocol):
    def some_method(self) -> None:
        ...

T = TypeVar("T", bound=SupportsSomeMethod)

def close_all(arg: T) -> None:
    arg.some_method()
```

Warning:
```
warning: TypeVar "T" appears only once in generic function signature (reportInvalidTypeVarUse)
```

**VS Code extension or command-line**
Command line utility version: `pyright 1.1.318`

**Comment by erictraut** [comment]
A TypeVar provides no value here. You can simply replace `arg: T` with `arg: SupportsSomeMethod`.

The warning is there because it is common for developers to misunderstand how TypeVars work and where they should be used and where they are not needed. ;)

**Comment by imLew** [comment]
Sorry, my bad, seems I'm indeed a common developer :sweat_smile: . I thought a TypeVar with a bound was needed to signal that things that aren't subtypes but do support the operation are accepted. 

Thanks for your quick response.

**Comment by erictraut** [comment]
If your object supports the operation, then it _is_ a subtype of the protocol. This is a form of "structural subtyping" as opposed to "nominal subtyping". If you're interested in more details, [PEP 483](https://peps.python.org/pep-0483/#subtype-relationships) is a good source of information.
